### PR TITLE
Fix unit test failure: TestNsenterExistsFile

### LINKED ISSUE #66809: TestNsenterExistsFile is failing
**Is this a BUG REPORT or FEATURE REQUEST?**: BUG REPORT

> Uncomment only one, leave it on its own line:
>
/kind bug
> /kind feature

**What happened**:
TestNsenterExistsFile test is failing during make test

**What you expected to happen**:
The test should have passed.

**How to reproduce it (as minimally and precisely as possible)**:
git clone https://github.com/kubernetes/kubernetes.git
run make test

**Anything else we need to know?**:
Fail logs:
```
ok  	k8s.io/kubernetes/pkg/util/metrics	0.004s
--- FAIL: TestNsenterExistsFile (0.01s)
	nsenter_mount_test.go:343: Test "simple non-accessible file": expected error, got none
	nsenter_mount_test.go:347: Test "simple non-accessible file": expected return value false, got true
E0731 01:45:42.561381   79766 mount_linux.go:495] format of disk "/dev/foo" failed: type:("ext4") target:("/tmp/mount336875516") options:(["defaults"])error:(formatting failed)
E0731 01:45:42.561927   79766 mount_linux.go:529] Could not determine if disk "/dev/foo" is formatted (exit 4)
FAIL
FAIL	k8s.io/kubernetes/pkg/util/mount	0.075s
ok  	k8s.io/kubernetes/pkg/util/net	0.002s
```

**Environment**:
- Kubernetes version (use `kubectl version`):
```
root@j:~/src/github.com/kubernetes/kubernetes# kubectl version
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.5", GitCommit:"32ac1c9073b132b8ba18aa830f46b77dcceb0723", GitTreeState:"clean", BuildDate:"2018-06-21T11:46:00Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
The connection to the server ip:6443 was refused - did you specify the right host or port?
root@j:~/src/github.com/kubernetes/kubernetes# 
```
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release): 
```
NAME="Ubuntu"
VERSION="16.04.4 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.4 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
```
- Kernel (e.g. `uname -a`):
`Linux j 4.4.0-130-generic #156-Ubuntu SMP Thu Jun 14 08:53:28 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by linyouchong**
**What this PR does / why we need it**:
Fix unit test failure: TestNsenterExistsFile

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #66809 

**Special notes for your reviewer**:
NONE

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/util/mount/nsenter_mount_test.go [hunk 1]
```diff
@@ -227,8 +228,26 @@ func TestNsenterExistsFile(t *testing.T) {
 
 				return path, nil
 			},
-			expectedOutput: false,
-			expectError:    true,
+			expectedOutput: func() bool {
+				user, err := user.Current()
```
Comment (jsafrane) [inline pkg/util/mount/nsenter_mount_test.go]: Can you detect that the user is root somewhere above (i.e. at the beginning of `TestNsenterExistsFile`) once and use it here?

```go
expectedOutput: isRoot, // ExistsPath succeeds when running as root
expectedError: !isRoot, // ExistsPath must fail when running as non-root
```

**Comment by linyouchong** [comment]
@jsafrane 
Fixed, Thank you!

**Pull request merged**